kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/DefaultExpressionEvaluator.java

```diff
@@ -276,10 +288,22 @@ ColumnVector visitOr(Or or) {
             PredicateChildrenEvalResult argResults = evalBinaryExpressionChildren(or);
             int numRows = argResults.rowCount;
             boolean[] result = new boolean[numRows];
-            boolean[] nullability = evalNullability(argResults.leftResult, argResults.rightResult);
+            boolean[] nullability = new boolean[numRows];
             for (int rowId = 0; rowId < numRows; rowId++) {
-                result[rowId] = argResults.leftResult.getBoolean(rowId) ||
-                    argResults.rightResult.getBoolean(rowId);
+                if (argResults.leftResult.isNullAt(rowId)) {
+                    // NULL || NULL --> NULL ; NULL || FALSE --> NULL ; NULL || TRUE --> TRUE
+                    nullability[rowId] = argResults.rightResult.isNullAt(rowId) ||
+                        !argResults.rightResult.getBoolean(rowId);
+                    result[rowId] = true;
+                } else if (argResults.rightResult.isNullAt(rowId)) {
+                    // TRUE || NULL --> TRUE ; FALSE || NULL --> NULL
+                    nullability[rowId] = !argResults.leftResult.getBoolean(rowId);
+                    result[rowId] = true;
```

Comment:
what does `result` hold? what does it being true mean?

particularly, what does `nullability[rowId]=true` and `result[rowId]=true` mean?

we should also clarify this here: https://github.com/delta-io/delta/blob/master/kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/vector/DefaultBooleanVector.java#L40

Reply:
where is this documented? it should be at the class-API level, and also inline in the code

Reply:
I can add comments and clarify the class docs